File: src/log/export/log_export.cpp
#include <export/cpp/export_macros.h>
#include <log/cpp/logger.h>

FLY_EXPORT_MODULE(_fly_log) {

FLY_EXPORT_ENUM(fly::LogLevel, "EXLogLevel")
    FLY_EXPORT_ENUM_VALUE("DEBUG", fly::LogLevel::DEBUG)
    FLY_EXPORT_ENUM_VALUE("INFO", fly::LogLevel::INFO)
    FLY_EXPORT_ENUM_VALUE("WARN", fly::LogLevel::WARN)
    FLY_EXPORT_ENUM_VALUE("ERROR", fly::LogLevel::ERROR);

FLY_EXPORT_FUNCTION("DBG", [](const fly::CMString& msg) {
    fly::Logger::instance()->log(fly::LogLevel::DEBUG, msg);
});

FLY_EXPORT_FUNCTION("INFO", [](const fly::CMString& msg) {
    fly::Logger::instance()->log(fly::LogLevel::INFO, msg);
});

FLY_EXPORT_FUNCTION("WARN", [](const fly::CMString& msg) {
    fly::Logger::instance()->log(fly::LogLevel::WARN, msg);
});

FLY_EXPORT_FUNCTION("ERR", [](const fly::CMString& msg) {
    fly::Logger::instance()->log(fly::LogLevel::ERROR, msg);
});

FLY_EXPORT_FUNCTION("flush_log", []() {
    fly::Logger::instance()->flush();
});

FLY_EXPORT_FUNCTION("init_log", [](const fly::CMString& base_dir, uint64_t worker_id) {
    fly::Logger::init(base_dir, worker_id);
});

FLY_EXPORT_FUNCTION("shutdown_log", []() {
    fly::Logger::shutdown();
});

FLY_EXPORT_FUNCTION("set_log_level", [](fly::LogLevel level) {
    fly::Logger::instance()->set_level(level);
});

}

File: src/log/log_export.py
"""log 模块 export 层——_fly_log.so 符号唯一导入点。

log 为纯 C++ 模块（无业务 Python 源码），本文件承载全部 Python 绑定
符号的导入；包根 ``from log import INFO`` 统一经此转发。全仓库含 C++
绑定的 Python 包均遵循此 export 导入层模式（docs/emir/lib-enhancement-plan.md §B）。
"""

from _fly_log import (
    DBG,
    INFO,
    WARN,
    ERR,
    EXLogLevel,
    flush_log,
    init_log,
    set_log_level,
    shutdown_log,
)
